File: spacetrader/front/src/app/model/response/player/crew-response.ts
import { PlanetResponse } from "../world/planet-response";
import { StarResponse } from "../world/star-response";
import { CargoItemResponse } from "./cargo-item-response";
import { PlayerResponse } from "./player-response";
import { SpaceShipResponse } from "./spaceship-response";

export interface CrewResponse{
    id: number
    crewMembers: PlayerResponse[]
    spaceship: SpaceShipResponse
    credits: number
    cargo: CargoItemResponse[]
    star: StarResponse
    planet: PlanetResponse
    actualCargoVolume: number
}

File: spacetrader/back/src/main/java/com/luisaq/spacetrader/dto/response/player/CrewResponse.java
package com.luisaq.spacetrader.dto.response.player;

import com.luisaq.spacetrader.dto.response.world.PlanetResponse;
import com.luisaq.spacetrader.dto.response.world.StarResponse;
import lombok.Builder;
import lombok.Data;

import java.math.BigDecimal;
import java.util.Set;

@Data
@Builder
public class CrewResponse {
    private Long id;
    private Set<PlayerResponse> crewMembers;
    private SpaceshipResponse spaceship;
    private BigDecimal credits;
    private Set<CargoItemResponse> cargo;
    private StarResponse star;
    private PlanetResponse planet;
    private Double actualCargoVolume;
}
